File: examples/process-file-path.js
#!/usr/bin/env node

/**
 * Example: Process a file from a specific path using the API
 *
 * Usage:
 *   node process-file-path.js /path/to/file.png
 *   node process-file-path.js /path/to/chat.txt deepseek
 *   node process-file-path.js /path/to/file.png claude group-id-123
 */

const axios = require('axios');

const API_URL = 'http://localhost:3000/api/screenshots/process-path';

async function processFile(filePath, provider = 'deepseek', groupId = null) {
  try {
    console.log(`Processing file: ${filePath}`);
    console.log(`Provider: ${provider}`);
    console.log('');

    const payload = {
      file_path: filePath,
      provider: provider,
    };

    if (groupId) {
      payload.group_id = groupId;
    }

    const response = await axios.post(API_URL, payload);

    console.log('Success! Result:');
    console.log(JSON.stringify(response.data, null, 2));

    return response.data;
  } catch (error) {
    console.error('Error processing file:');
    if (error.response) {
      console.error('Status:', error.response.status);
      console.error('Error:', error.response.data);
    } else {
      console.error(error.message);
    }
    process.exit(1);
  }
}

// Parse command line arguments
const args = process.argv.slice(2);

if (args.length === 0) {
  console.log('Usage: node process-file-path.js <file_path> [provider] [group_id]');
  console.log('');
  console.log('Examples:');
  console.log('  node process-file-path.js /path/to/screenshot.png');
  console.log('  node process-file-path.js /path/to/chat.txt deepseek');
  console.log('  node process-file-path.js /path/to/screenshot.png claude group-id-123');
  process.exit(1);
}

const [filePath, provider, groupId] = args;

processFile(filePath, provider, groupId);

File: examples/process-file-path.py
#!/usr/bin/env python3

"""
Example: Process a file from a specific path using the API

Usage:
  python process-file-path.py /path/to/file.png
  python process-file-path.py /path/to/chat.txt deepseek
  python process-file-path.py /path/to/file.png claude group-id-123
"""

import sys
import json
import requests

API_URL = 'http://localhost:3000/api/screenshots/process-path'

def process_file(file_path, provider='deepseek', group_id=None):
    """Process a file using the API"""

    print(f"Processing file: {file_path}")
    print(f"Provider: {provider}")
    print()

    payload = {
        'file_path': file_path,
        'provider': provider,
    }

    if group_id:
        payload['group_id'] = group_id

    try:
        response = requests.post(API_URL, json=payload)
        response.raise_for_status()

        print("Success! Result:")
        print(json.dumps(response.json(), indent=2))

        return response.json()

    except requests.exceptions.RequestException as e:
        print(f"Error processing file: {e}")
        if hasattr(e, 'response') and e.response is not None:
            print(f"Status: {e.response.status_code}")
            print(f"Error: {e.response.text}")
        sys.exit(1)

def main():
    if len(sys.argv) < 2:
        print("Usage: python process-file-path.py <file_path> [provider] [group_id]")
        print()
        print("Examples:")
        print("  python process-file-path.py /path/to/screenshot.png")
        print("  python process-file-path.py /path/to/chat.txt deepseek")
        print("  python process-file-path.py /path/to/screenshot.png claude group-id-123")
        sys.exit(1)

    file_path = sys.argv[1]
    provider = sys.argv[2] if len(sys.argv) > 2 else 'deepseek'
    group_id = sys.argv[3] if len(sys.argv) > 3 else None

    process_file(file_path, provider, group_id)

if __name__ == '__main__':
    main()
